servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -37,37 +125,195 @@
    */
   public static void main(String[] args) {
     checkArgs(args);
-    BlockWorker worker = null;
-    FileSystemWorker fileWorker = null;
-
+    TachyonWorker worker;
     try {
-      worker = new BlockWorker();
-       // Setup the file worker
-      LOG.info("Started file system worker at worker with id {}", WorkerIdRegistry.getWorkerId());
-      fileWorker = new FileSystemWorker(worker.getBlockDataManager());
-
+      worker = new TachyonWorker();
+      worker.start();
     } catch (Exception e) {
-      LOG.error("Failed to initialize the block worker, exiting.", e);
+      LOG.error("Uncaught exception while running worker, shutting down and exiting.", e);
       System.exit(-1);
     }
+  }
 
-    try {
-      // Start the file system worker
-      fileWorker.start();
-      worker.process();
+  /**
+   * @return the worker RPC service bind host
+   */
+  public String getRPCBindHost() {
+    return NetworkAddressUtils.getThriftSocket(mThriftServerSocket).getInetAddress()
+        .getHostAddress();
+  }
+
+  /**
+   * @return the worker RPC service port
+   */
+  public int getRPCLocalPort() {
+    return mRPCPort;
+  }
+
+  /**
+   * @return the worker data service bind host (used by unit test only)
+   */
+  public String getDataBindHost() {
```

Comment:
it is not decided yet whether dataserver will just belong to blockworker or not. very likely we will move dataserver from block worker to tachyon worker.